cartoframes/data/observatory/catalog/repository/dataset_repo.py

```diff
@@ -18,10 +19,19 @@ def __init__(self):
         super(DatasetRepository, self).__init__(_DATASET_ID_FIELD, _ALLOWED_FILTERS, _DATASET_SLUG_FIELD)
 
     def get_all(self, filters=None, credentials=None):
+        if credentials is not None:
```

Comment:
This should be extracted to utils and be used like: `filters = add_subscription_ids(filters, credentials, DATASET_TYPE)`. The same in geography_repo.